openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -314,7 +314,9 @@ For security considerations, see (#credential-offer-security).
 This specification defines the following parameters for the Credential Offer object:
 
 * `credential_issuer`: REQUIRED. The URL of the Credential Issuer, as defined in (#credential-issuer-identifier), from which the Wallet is requested to obtain one or more Credentials. The Wallet uses it to obtain the Credential Issuer's Metadata following the steps defined in (#credential-issuer-wellknown).
-* `credentials`: REQUIRED. A JSON array, where every entry is a JSON object or a JSON string. If the entry is an object, the object contains the data related to a certain credential type the Wallet MAY request. Each object MUST contain a `format` Claim determining the format of the credential to be requested and further parameters characterizing the type of the credential to be requested as defined in (#format_profiles). If the entry is a string, the string value MUST be one of the `scope` values in one of the objects in the `credentials_supported` Credential Issuer metadata parameter. When processing, the Wallet MUST resolve this string value to the respective object.
+* `credentials`: REQUIRED. A JSON array, where every entry is a JSON string or a JSON object describing a credential the Wallet MAY request. If the entry is a string, the string value MUST be one of the `scope` values included in a `credentials_supported` Credential Issuer metadata parameter as defined in (#credential-metadata-object). When processing, the Wallet MUST resolve this string value to the respective object. If the entry is an object, the object contains the data related to a certain credential type. Each object contains the following parameters: 
+  * `format`: REQUIRED. JSON string determining the format of the credential.
+  * Parameters characterizing the type of the credential to be requested: REQUIRED. These parameters are specific to the credential format profile, some of which are defined in (#format_profiles).
```

Comment:
The current definition does not include the `display` and other properties defined in (#credential-metadata-object) — is that intentional?

Reply:
Credential Offer is a mere offer as explained in the spec already and elaborated in PR #73. 

@danielfett @pmhsfelix `display`, `cryptographic_binding_methods_supported` and `cryptographic_suites_supported` should be discovered using credential issuer metadata instead of being passed in the credential offer.

Reply:
Does that mean that an offer should always be associated to exactly one entry in the metadata's `credentials_supported` array?
Namely, can't we use offers for credentials _kinds_ that are not statically described in the server's metadata?

Reply:
> can't we use offers for credentials kinds that are not statically described in the server's metadata?

in my mental model issuer was offering concrete credentials using the offer. what does offering `credential kinds` mean? how does a wallet know what to request?

Reply:
I used the term "kind" to avoid the term "type", but I'm also thinking in the offer describing a concrete credential.
In my view, the wallet needs:
- display info to show the offered credential to the end user. When using W3C data model, this can be a `credentialSubject`.
- cryptographic parameters needed for the credential request proof (e.g. `cryptographic_binding_methods_supported`).
- The *identifier*, to be sent on the authorization request or token request, when using pre-authorized code.
This seems enough for the wallet to:
1) Present a UI to the user.
2) Know how to perform the complete flow.

The offer should be self-sufficient for the wallet to perform a full credential request flow, without needing to use issuer's metadata `credentials_supported`.
The authorization server ID and credential endpoint URL could still come from the issuer metadata.

Reply:
I am less sure about the need to pass self-containing credential description in the credential offer - I think that would require signing credential offer because credential offer is fundamentally not trusted, which I think should be avoided. 
You seem to have concern around "needing to use issuer's metadata credentials_supported", but I am not sure I fully understant what is it? 
would enabling a new option, where instead of fetching entire issuer metadata and searching within it itself, the wallet can send a request to the issuer using the identifier it has obtained from the credential offer so that the issuer can return only the part of the credentials_supported object that the wallet needs? @pmhsfelix

Reply:
I think no need for a new issue. discussion is happening in issue #77

Reply:
also I think issue #42 might be related.

Reply:
this topic is out of scope for this PR, I will revert this change. I was just trying to clarify the text, so my changes here did not introduce any normative changes.